compiler/crates/relay-codegen/src/build_ast.rs

```diff
@@ -1850,64 +1850,64 @@ impl<'schema, 'builder, 'config> CodegenBuilder<'schema, 'builder, 'config> {
+                    key: CODEGEN_CONSTANTS.id,
+                    value: Primitive::RawString(id.clone()),
+                },
+                QueryID::External(module_name) => ObjectEntry {
+                    key: CODEGEN_CONSTANTS.id,
+                    value: Primitive::JSModuleDependency(JSModuleDependency {
                         path: *module_name,
                         named_import: None,
                         import_as: None,
-                    })
-                }
-                None => Primitive::Null,
-            },
-        };
-
-        let mut params_object = if let Some(text) = request_parameters.text {
-            vec![
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.cache_id,
-                    value: Primitive::RawString(md5(&text)),
-                },
-                id_prop,
-                metadata_prop,
-                name_prop,
-                operation_kind_prop,
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.text,
-                    value: Primitive::RawString(text),
-                },
-            ]
-        } else if request_parameters.id.is_some() {
-            vec![
-                id_prop,
-                metadata_prop,
-                name_prop,
-                operation_kind_prop,
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.text,
-                    value: Primitive::Null,
+                    }),
                 },
-            ]
+            }
         } else {
-            vec![
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.cache_id,
-                    value: Primitive::RawString(md5(operation.name.item.0.lookup())),
-                },
-                id_prop,
-                metadata_prop,
-                name_prop,
-                operation_kind_prop,
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.text,
-                    value: Primitive::Null,
-                },
-            ]
+            ObjectEntry {
+                key: CODEGEN_CONSTANTS.id,
+                value: Primitive::Null,
+            }
         };
 
+        let mut params_object = vec![];
+
+        match request_parameters.request_format {
+            Some(
+                RequestFormat::Text(Some(ref text)) | RequestFormat::Migration(_, Some(ref text)),
+            ) => params_object.push(ObjectEntry {
+                key: CODEGEN_CONSTANTS.cache_id,
+                value: Primitive::RawString(md5(&text.clone())),
+            }),
+            None => params_object.push(ObjectEntry {
+                key: CODEGEN_CONSTANTS.cache_id,
+                value: Primitive::RawString(md5(operation.name.item.0.lookup())),
+            }),
+            _ => (),
+        }
+
+        params_object.push(id_prop);
+        params_object.push(metadata_prop);
+        params_object.push(name_prop);
+        params_object.push(operation_kind_prop);
```

Comment:
Could we unconditionally initialize the params object with the ones we know we are going to include and then just have one push for the request_format case?

Reply:
I've done it like this in order to maintain the existing artifact ordering. If I change this, it will invalidate all existing artifacts (even without this feature on). If that's okay to for you at Meta, I can do this - but wanted to reduce your churn